File: Sqrilizz-Reports-Lite/src/main/java/dev/sqrilizz/reports/WebhookManager.java
package dev.sqrilizz.reports;

import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;

/**
 * Simple webhook notifications
 */
public class WebhookManager {
    
    private final Main plugin;
    private final HttpClient httpClient;
    private final String webhookUrl;
    private final boolean enabled;
    
    public WebhookManager(Main plugin) {
        this.plugin = plugin;
        this.httpClient = HttpClient.newBuilder()
            .connectTimeout(Duration.ofSeconds(10))
            .build();
        
        this.webhookUrl = plugin.getConfig().getString("webhook.url", "");
        this.enabled = !webhookUrl.isEmpty();
        
        if (enabled) {
            plugin.getLogger().info("Webhook notifications enabled");
        }
    }
    
    /**
     * Send report notification via webhook
     */
    public void sendReport(ReportManager.Report report) {
        if (!enabled) return;
        
        try {
            String json = createReportJson(report);
            sendWebhook(json);
            
        } catch (Exception e) {
            plugin.getLogger().warning("Failed to send webhook: " + e.getMessage());
        }
    }
    
    /**
     * Create JSON payload for report
     */
    private String createReportJson(ReportManager.Report report) {
        return String.format("""
            {
                "event": "report_created",
                "report": {
                    "id": %d,
                    "timestamp": %d,
                    "reporter": "%s",
                    "target": "%s",
                    "reason": "%s",
                    "location": "%s",
                    "resolved": false
                },
                "server": "%s"
            }
            """,
            report.getId(),
            report.getTimestamp(),
            escapeJson(report.getReporterName()),
            escapeJson(report.getTargetName()),
            escapeJson(report.getReason()),
            escapeJson(report.getLocation()),
            escapeJson(plugin.getServer().getName())
        );
    }
    
    /**
     * Send webhook HTTP request
     */
    private void sendWebhook(String json) throws IOException, InterruptedException {
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(webhookUrl))
            .header("Content-Type", "application/json")
            .header("User-Agent", "Sqrilizz-Reports-Lite/8.0")
            .POST(HttpRequest.BodyPublishers.ofString(json))
            .timeout(Duration.ofSeconds(30))
            .build();
        
        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
        
        if (response.statusCode() >= 400) {
            throw new IOException("Webhook returned status: " + response.statusCode());
        }
    }
    
    /**
     * Escape JSON special characters
     */
    private String escapeJson(String text) {
        if (text == null) return "";
        return text.replace("\\", "\\\\")
                  .replace("\"", "\\\"")
                  .replace("\n", "\\n")
                  .replace("\r", "\\r")
                  .replace("\t", "\\t");
    }
    
    public boolean isEnabled() {
        return enabled;
    }
}

File: Sqrilizz-Reports-Lite/src/main/kotlin/dev/sqrilizz/reports/WebhookManager.kt
package dev.sqrilizz.reports

import java.net.URI
import java.net.http.HttpClient
import java.net.http.HttpRequest
import java.net.http.HttpResponse
import java.time.Duration

class WebhookManager(private val plugin: Main) {

    private val httpClient: HttpClient = HttpClient.newBuilder()
        .connectTimeout(Duration.ofSeconds(10))
        .build()

    private val webhookUrl: String = plugin.config.getString("webhook.url", "")!!
    val isEnabled: Boolean = webhookUrl.isNotEmpty()

    init {
        if (isEnabled) {
            plugin.logger.info("Webhook notifications enabled")
        }
    }

    fun sendReport(report: ReportManager.Report) {
        if (!isEnabled) return

        try {
            val json = createReportJson(report)
            sendWebhook(json)
        } catch (e: Exception) {
            plugin.logger.warning("Failed to send webhook: ${e.message}")
        }
    }

    private fun createReportJson(report: ReportManager.Report): String = """
        {
            "event": "report_created",
            "report": {
                "id": ${report.id},
                "timestamp": ${report.timestamp},
                "reporter": "${escapeJson(report.reporterName)}",
                "target": "${escapeJson(report.targetName)}",
                "reason": "${escapeJson(report.reason)}",
                "location": "${escapeJson(report.location)}",
                "resolved": false
            },
            "server": "${escapeJson(plugin.server.name)}"
        }
    """.trimIndent()

    private fun sendWebhook(json: String) {
        val request = HttpRequest.newBuilder()
            .uri(URI.create(webhookUrl))
            .header("Content-Type", "application/json")
            .header("User-Agent", "Sqrilizz-Reports-Lite/${plugin.description.version}")
            .POST(HttpRequest.BodyPublishers.ofString(json))
            .timeout(Duration.ofSeconds(30))
            .build()

        val response = httpClient.send(request, HttpResponse.BodyHandlers.ofString())
        if (response.statusCode() >= 400) {
            throw java.io.IOException("Webhook returned status: ${response.statusCode()}")
        }
    }

    private fun escapeJson(text: String?): String =
        (text ?: "")
            .replace("\\", "\\\\")
            .replace("\"", "\\\"")
            .replace("\n", "\\n")
            .replace("\r", "\\r")
            .replace("\t", "\\t")
}
